### PR TITLE
[4.0] adapting saved language message to different settings

### PR DESCRIPTION
**Pull request by bahl24**
Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

Pull Request for Issue #23850  .

### Summary of Changes
Added style for class=red in templates/cassiopeia/scss/blocks/_alerts.scss
Required for administrator/language/en-GB/en-GB.com_languages.ini.

### CODE REVIEW

**Comment by bahl24** [comment]
Not sure if we need a red background as in joomla-alert type=danger, as in an error alert

**Comment by infograf768** [comment]
I suggest to rather modify the string instead of adding a red colour as the class red in the string is a remnant of 2.5
I.e. Change
`COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED="Default Language Saved. This does not affect users that have chosen a specific language on their profile or on the login page.<br><strong class=\"red\">Warning!</strong> When using the multilingual functionality (ie when the plugin System - Language Filter is enabled) the Site Default Language also has to be a published Content language."
`
to
`COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED="Default Language Saved. This does not affect users that have chosen a specific language on their profile or on the login page.<br><strong>Warning!</strong> When using the multilingual functionality (ie when the plugin System - Language Filter is enabled) the Site Default Language also has to be a published Content language."`

**Comment by kofaysi** [comment]
Shouldn't it be two messages (two strings) in fact? One us about successful action "save". The other is the "multilingual" warning or notice. I think a warning with a green (as in valid, succesfull, OK) is just obscure.

Can multiple support be detected programmatically?

**Comment by infograf768** [comment]
> Can multiple support be detected programmatically?

yes.
We can modify `/administrator/components/com_languages/Controller/InstalledController.php`
From line 53-54
```
			$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
			$type = 'message';
```
to
```
			if (Multilanguage::isEnabled())
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_SAVED');
				$type = 'message';
			}
			else
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
				$type = 'message';
			}
```
Then modify the string
`COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED="Default Language Saved. This does not affect users that have chosen a specific language on their profile or on the login page."`

And add a new one with a modified former text value.

`"COM_LANGUAGES_MSG_DEFAULT_MULTILANG_SAVED="Default Language Saved. This does not affect users that have chosen a specific language on their profile or on the login page.<br><strong>Warning!</strong> The Site Default Language also has to be a published Content language."`


Imho, the Warning in bold is sufficient.

**Comment by bahl24** [comment]
Ok @infograf768 @kofaysi let me work on this and make the changes you suggested

**Comment by kofaysi** [comment]
I would go for (inspired by the code by @infograf768 )
```
			$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
			$type = 'message';

			if (Multilanguage::isEnabled())
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_WARNING');
				$type = 'notice'; //or warning?
			}
```
Then modify the string
`COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED="Default Language Saved. This does not affect users that have chosen a specific language on their profile or on the login page."`

And add a new one with a modified former text value.

`"COM_LANGUAGES_MSG_DEFAULT_MULTILANG_WARNING="The Site Default Language also has to be a published Content language."`

1. Separating the two messages makes it easier to manipulate any of them in the future.
2. It clearly distinguishes a message and a warning from each other.

mic drop :) I'll let you work, dear @bahl24

**Comment by infograf768** [comment]
@kofaysi 
Your proposal will not work. Only one message will be displayed and it will be the second one when site is multilang.

EDIT: the way to do it would rather be
```
			if (Multilanguage::isEnabled())
			{
				$this->app->enqueueMessage(Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_WARNING'), 'warning');
			}
```
Possible not nice UI: 2 messages are displayed and have to be closed.

File: administrator/components/com_languages/Controller/InstalledController.php [hunk 1]
```diff
@@ -50,8 +50,17 @@ public function setDefault()
...
+			{
+				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_SAVED');
+				$type = 'message';
+			}
+			else
+			{
+				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
+				$type = 'message';
+			}
+
```
Comment (Quy) [inline administrator/components/com_languages/Controller/InstalledController.php]:
```suggestion
```

**Comment by bahl24** [comment]
@Quy removed extra spaces, kindly review.

**Comment by bahl24** [comment]
@infograf768 @Quy Kindly test this

**Comment by infograf768** [comment]
@bahl24 
1. You need to add:
`use Joomla\CMS\Language\Multilanguage;`
after
`use Joomla\CMS\Language\Language;` line 16

2. Also, I suggest to differentiate admin default language change and site language default change by using this code as the warning concerning content language is not necessary for admin
```
			if (Multilanguage::isEnabled())
			{
				if ($model->getState('client_id') == 0)
				{
					$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_SAVED');
					$type = 'message';
				}
				else
				{
					$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
					$type = 'message';
				}
			}
			else
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
				$type = 'message';
			}
```

**Comment by Quy** [comment]
This can be simplified as follows:
```
			if (Multilanguage::isEnabled() && $model->getState('client_id') == 0)
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_MULTILANG_SAVED');
				$type = 'message';
			}
			else
			{
				$msg = Text::_('COM_LANGUAGES_MSG_DEFAULT_LANGUAGE_SAVED');
				$type = 'message';
			}
```

**Comment by bahl24** [comment]
@Quy @infograf768 @kofaysi I think the changes are finalized, kindly test them.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on fb7f3771757f2bb46ddbc8c27885ce5d316af821<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23851">issues.joomla.org/tracker/joomla-cms/23851</a>.</sub>

**Comment by bahl24** [comment]
@Quy Kindly test

**Comment by infograf768** [comment]
restarted drone

**Comment by bahl24** [comment]
@infograf768 I think all modifications are now done, restarted drone for CI tests

File: administrator/components/com_languages/Controller/InstalledController.php [hunk 2]
```diff
@@ -50,8 +51,17 @@ public function setDefault()
...
```
Comment (Quy) [inline administrator/components/com_languages/Controller/InstalledController.php]: Remove tabs.

Comment (bahl24) [reply administrator/components/com_languages/Controller/InstalledController.php]: Are you talking about trailing whitespace or to remove the new line?

Comment (Quy) [reply administrator/components/com_languages/Controller/InstalledController.php]: Trailing whitespace

Comment (Quy) [reply administrator/components/com_languages/Controller/InstalledController.php]: Actually delete the new line.

**Comment by bahl24** [comment]
@Quy Kindly test

**Comment by infograf768** [comment]
@wilsonge
Please merge.

**Comment by wilsonge** [comment]
LGTM! Thanks!

**Pull request merged**